### PR TITLE
Return non-zero on travis if tests fail

### PR DESCRIPTION
**Pull request by sbc100**
Oops, we were silently ignoring tests failures on travis.

Also turn on -v when running travis tests so it records the tests it runs

### CODE REVIEW

File: scripts/travis-test.sh [hunk 1]
```diff
@@ -18,17 +18,19 @@
...
 set -o errexit
 
+RTN=0
 SCRIPT_DIR="$(cd "$(dirname "$0")"; pwd -P)"
 source "${SCRIPT_DIR}/travis-common.sh"
 
 log_and_run() {
   echo $*
-  exec $*
+  (exec $*) && true
```
Comment (pipcet) [inline scripts/travis-test.sh]: Probably a silly question, but what is the `&& true` supposed to do here? I'm no Travis expert, but in a Bourne shell that's just a nop, right?

Comment (binji) [inline scripts/travis-test.sh]: My understanding is that this runs the command in a subshell and captures the exit code, even though this script is running in error exit mode. That way we can keep running tests even if one of them fails.

Comment (pipcet) [inline scripts/travis-test.sh]: Thanks! I think the whole line is equivalent to just `$*`, but if it works with Travis, it's probably best not to fiddle with it :-)

The new code looks correct to me and appears to fix the bug, so this was mere curiosity.

File: test/binary/relocs.txt [hunk 2]
```diff
@@ -17,4 +17,5 @@ section("reloc.name") {
 (;; STDOUT ;;;
 (module
   (type (;0;) (func (result i32))))
+XXXXX
```
Comment (pipcet) [inline test/binary/relocs.txt]: Er, we don't want to merge this bit, right?

Comment (sbc100) [inline test/binary/relocs.txt]: Exactly. I'm just using this to verify travis will fail on a failing test (which it doesn't right  now!)

Comment (pipcet) [inline test/binary/relocs.txt]: Thanks again. Good catch, by the way!

File: scripts/travis-test.sh [hunk 3]
```diff
@@ -53,6 +59,10 @@ for BUILD_TYPE in ${BUILD_TYPES_UPPER}; do
   else
     if set_run_test_args ${COMPILER} ${BUILD_TYPE}; then
       run_tests
+      break
```
Comment (binji) [inline scripts/travis-test.sh]: why break?

Comment (sbc100) [inline scripts/travis-test.sh]: oops, reverted

File: scripts/travis-test.sh [hunk 4]
```diff
@@ -18,17 +18,23 @@
...
+  if ! $*; then
+    echo "travis-test.sh: sub-command failed: $*"
+    RTN=1;
+  fi
 }
 
 run_tests() {
-  (cd ${ROOT_DIR} && log_and_run test/run-tests.py --bindir ${BINDIR} $* --timeout=10) && true
-  (log_and_run ${BINDIR}/wabt-unittests) && true
+  log_and_run test/run-tests.py -v --bindir ${BINDIR} $* --timeout=10
```
Comment (binji) [inline scripts/travis-test.sh]: Can you mention that you turned on verbosity in the commit message too?

**Pull request merged**